src/Uno.UI/UI/Xaml/ResourceResolver.cs

```diff
@@ -0,0 +1,323 @@
+
+		private static readonly ILogger _log = typeof(ResourceResolver).Log();
+
+		private static readonly Stack<XamlScope> _scopeStack;
+
+		/// <summary>
+		/// The current xaml scope for resource resolution.
+		/// </summary>
+		internal static XamlScope CurrentScope => _scopeStack.Peek();
+
+		static ResourceResolver()
+		{
+			_scopeStack = new Stack<XamlScope>();
+			_scopeStack.Push(XamlScope.Create()); //There should always be a base-level scope (this will be used when no template is being resolved)
+		}
+
+		/// <summary>
+		/// Performs a one-time, typed resolution of a named resource, using Application.Resources.
+		/// </summary>
+		/// <returns></returns>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public static T ResolveResourceStatic<T>(object key, object context = null)
+		{
+			if (TryStaticRetrieval(key, context, out var value) && value is T tValue)
+			{
+				return tValue;
+			}
+
+			return default(T);
+		}
+
+		/// <summary>
+		/// Retrieve a resource from top-level resources (Application-level and system level).
+		/// </summary>
+		/// <typeparam name="T">The expected resource type</typeparam>
+		/// <param name="key">The resource key to search for</param>
+		/// <param name="fallbackValue">Fallback value to use if no resource is found</param>
+		/// <returns>The resource, is one of the given key and type is found, else <paramref name="fallbackValue"/></returns>
+		/// <remarks>
+		/// Use <see cref="ResolveTopLevelResource{T}(object, T)"/> when user-defined Application-level values should be considered (most
+		/// of the time), use <see cref="GetSystemResource{T}(object)"/> if they shouldn't
+		/// </remarks>
+		internal static T ResolveTopLevelResource<T>(object key, T fallbackValue = default)
+		{
+			if (TryTopLevelRetrieval(key, context: null, out var value) && value is T tValue)
+			{
+				return tValue;
+			}
+
+			return fallbackValue;
+		}
+
+		/// <summary>
+		/// Apply a StaticResource or ThemeResource assignment to a DependencyProperty of a DependencyObject. The assignment will be provisionally
+		/// made immediately using Application.Resources if possible, and retried at load-time using the visual-tree scope.
+		/// </summary>
+		/// <param name="owner">Owner of the property</param>
+		/// <param name="property">The property to assign</param>
+		/// <param name="resourceKey">Key to the resource</param>
+		/// <param name="context">Optional parameter that provides parse-time context</param>
+		[EditorBrowsable(EditorBrowsableState.Never)]
+		public static void ApplyResource(DependencyObject owner, DependencyProperty property, object resourceKey, bool isThemeResourceExtension, object context = null)
+		{
+			// Set initial value based on statically-available top-level resources.
+			if (TryStaticRetrieval(resourceKey, context, out var value))
+			{
+				owner.SetValue(property, BindingPropertyHelper.Convert(() => property.Type, value));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Refactor this method to add validation of parameter 'owner' before using it.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)